.github/workflows/composites/upload-docker-images/action.yaml

```diff
@@ -0,0 +1,15 @@
+# we are not going to use upload/download actions, since it is slower than cache usage
```

Comment:
I am using github composite actions, as it keeps the code much more understandable and manageable.